src/app/app.component.html

```diff
@@ -25,11 +31,23 @@
   Loading...
 </div>
 
+<div class="overlay-container loading-container drop-zone-container darker"
+     *ngIf="(ebookDisplayManagerService.loadingFiles$ | async)">
+  <div class="progress-container">
+    <div class="truncate">{{ebookDisplayManagerService.loadingFiles$.value?.title}}</div>
+    <div class="progress">
+      <span [style.width]="ebookDisplayManagerService.loadingFiles$.value?.progress"></span>
+    </div>
+  </div>
+</div>
+
 <ng-template #notLoadingRef>
-  <label class="overlay-container drop-zone-container label-button" *ngIf="!(visible$ | async)">
-    Drop or select a HTMLZ or EPUB file to continue
-    <input type="file" accept=".htmlz,.epub" (change)="onInputChange(inputRef)" hidden #inputRef>
+  <label class="overlay-container drop-zone-container label-button" *ngIf="!(visible$ | async)" (click)="zoneInputRef.click();"
+    (contextmenu)="$event.preventDefault();zoneInputDirRef.click();">
+    {{dropZoneLabel}}
   </label>
+  <input type="file" accept=".htmlz,.epub" multiple hidden #zoneInputRef (change)="onInputChange(zoneInputRef)">
+  <input type="file" webkitdirectory directory multiple hidden #zoneInputDirRef (change)="onInputChange(zoneInputDirRef)">
```

Comment:
Nope does not work.. actually the directory attribute is the one which does not work as of today. i kept as in case vendors will implement it but for now the webkit one seems to be required (else it falls back to the files selection)